test/test_nn.py

```diff
@@ -2089,6 +2089,41 @@ def test_embedding_sparse_backward(self):
         self.assertEqual(embedding.weight.grad._indices(), torch.LongTensor([[7, 1, 3, 8, 1, 3]]))
         self.assertEqual(embedding.weight.grad._values(), torch.tensor(1.).expand(6, 3))
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_embedding_sparse_half_backward(self):
+        # same as test_embedding_sparse_backward above but testing half types in
```

Comment:
Since this test looks almost exactly the same, could we refactor it so that test_embedding_sparse_backward and this test share more code?

Reply:
good point, started with a copy paste to get it working and forgot to simplify.